libs/server/Resp/RespServerSession.cs

```diff
@@ -883,16 +889,19 @@ private void WriteDirectLarge(ReadOnlySpan<byte> src)
                 src = src.Slice(destSpace);
 
                 // Send and reset output buffer
-                Send(networkSender.GetResponseObjectHead());
+                Send(networkSender.GetResponseObjectHead(), true);
                 networkSender.GetResponseObject();
                 dcurr = networkSender.GetResponseObjectHead();
                 dend = networkSender.GetResponseObjectTail();
             }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void Send(byte* d)
+        private void Send(byte* d, bool isChunked = false)
```

Comment:
We call `SendAndReset` in many places after partial command processing, and those are not caught by this code change. It seems it can be made correct by moving the setting to false of waitForAofBlocking, back to the finally in `TryConsumeMessages`.

But unfortunately it is a bit conservative, because we might have 100 commands in the batch, the first command made waitForAofBlocking true and we finished flushing its result after blocking, yet the bool will continue to stay true and block the AOF flush for all the remaining 99 commands even if they did not need to block.